pkg/networkservice/common/heal/client.go

```diff
@@ -162,13 +153,25 @@ func (f *healClient) Request(ctx context.Context, request *networkservice.Networ
 	return rv, nil
 }
 
+func (f *healClient) Stop() {
```

Comment:
Why Stop() vs a finalizer ?

Reply:
Probably that doesn't make sense for real life executions, but this is sufficient for tests. Test finished and printed "PASS" line, but resource's finalization just started and prints some additional lines after. That lead to some collisions on running test with `count=10000` parameter - it just fails after several thousands retries.

Reply:
Could you say more, I don't quite follow...

Reply:
To track some concurrency issues I ran test with parameter `-test.count 10000`. And runtime.Finalizerer hooks don't work between separate executions of test. So the number of gorotines grows because they continue to hang on channels. That's easy to track using `runtime.NumGoroutine()` it equal 120002 after 10000 execution of the same test. 

If we run test once there are also might be issues: Finalizer not always manages to work until test ends, but if it does then we will see some artifacts already after "PASS" line:
```
=== RUN   TestHealClient_New
=== RUN   TestHealClient_New/initial_state_transfer
--- PASS: TestHealClient_New (0.00s)
    --- PASS: TestHealClient_New/initial_state_transfer (0.00s)
PASS
time="2020-02-01T23:58:08+07:00" level=info msg="error rpc error: code = Canceled desc = grpc: the client connection is closing"
``` 

Probably all that issues doesn't worth adding `Stop` method

Reply:
Are we still leaking goroutines after the serializer fix?

Reply:
Fixed with per-chain Context